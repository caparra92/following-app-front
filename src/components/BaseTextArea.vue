<template>
    <ion-row>
      <ion-col>
        <ion-item class="container">
          <textarea
            v-bind="$attrs"
            :placeholder="label"
            :value="modelValue"
            :cols="col"
            :rows="row"
            class="field"
            @input="$emit('update:modelValue', $event.target.value)"
          >
          </textarea>
        </ion-item>
      </ion-col>
    </ion-row>
  </template>
  <script setup>
  import { IonRow, IonCol, IonItem } from '@ionic/vue';
  
  const props = defineProps({
    label: {
      type: String,
      default: "",
    },
    modelValue: {
      type: [String, Number],
      default: "",
    },
    col: {
      type: String,
      default: "10",
    },
    row: {
      type: String,
      default: "5",
    }
  });
  </script>
  <style scoped>
  .container {
    width: 100%;
    padding: 0;
    margin: 0;
  }

  textarea {
    resize: none;
  }
  </style>
  