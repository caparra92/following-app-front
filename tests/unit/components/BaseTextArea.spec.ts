import { mount, VueWrapper } from "@vue/test-utils";
import BaseTextArea from "@/components/BaseTextArea.vue";

describe('BaseTextArea tests', () => {

    let wrapper: VueWrapper;

    beforeEach(() => {
        wrapper = mount(BaseTextArea, {
            props: {
                label: 'Category',
                modelValue: 1500,
                col: '10',
                row: '5'
            }   
        });
    });

    test('Should match with snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('Should verify that component exists', () => {
        expect(wrapper.exists()).toBeTruthy();
    });

    test('Should have <ion-item/> tag with class = "icon"', () => {
        const ionItem = wrapper.findComponent('ion-item');
        expect(ionItem.attributes().class).toBe("container");
    });

    test('Should have <textarea/> tag with props label, modelValue, col and row', () => {
        expect(wrapper.props().label).toBe('Category');
        expect(wrapper.props().modelValue).toBe(1500);
        expect(wrapper.props().col).toBe('10');
        expect(wrapper.props().row).toBe('5');
    });
});