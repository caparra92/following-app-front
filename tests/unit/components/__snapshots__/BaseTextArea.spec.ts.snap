// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseTextArea tests Should match with snapshot 1`] = `
<ion-row class="">
  <ion-col class="">
    <ion-item class="container"><textarea placeholder="Category" cols="10" rows="5" class="field">
          </textarea></ion-item>
  </ion-col>
</ion-row>
`;
